src/settings.rs

```diff
@@ -231,12 +261,26 @@ mod tests {
     #[test]
     fn test_default_config() {
         let key = "RUN_MODE";
-        env::set_var(key, "TEST");
+        env::set_var(key, "development");
 
         // Read config
         let config = Settings::new(None).unwrap();
 
         // Validate
         config.validate().unwrap();
+        assert!(config.found_config_files)
+    }
+
+    #[test]
+    fn test_no_config_files() {
+        let non_existing_config_path = "config/non_existing_config".to_string();
+        env::remove_var("RUN_MODE");
```

Comment:
@generall this combined with the `env::set_var` in the above test cause a race condition.

Reply:
Correct. Environment variables are a shared resources, and tests can be run concurrently.

I've added a test that ensures the embedded configuration can be loaded on it's own. I think that's the best we can do without cluttering the `Settings` interface.

Reply:
happened several times so I have created https://github.com/qdrant/qdrant/issues/1803